tests/tests_service.yml

```diff
@@ -0,0 +1,313 @@
+- name: Test firewall user defined services
+  hosts: all
+  become: true
+  tasks:
+    - name: Test firewall user defined services
+      block:
+
+        # INIT TEST
+
+        - name: Start from clean slate
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+              - previous: replaced
+
+        # Verify service states are as intended
+
+        - name: get all services
+          command: firewall-cmd --permanent --get-services
+          register: result
+
+        - name: Fail if http not present after reset
+          fail:
+            msg: Reset default resulted in unexpected state
+          when: result.stdout.find("http ") == -1
+
+        - name: Fail if systemroletest present prior to test run
+          fail:
+            msg: test service present after reseting defaults
+          when: result.stdout.find("systemroletest") != -1
```

Comment:
you can use `command` here too, but use `register` and `failed_when` 
```yaml
        - name: Fail if http not present after reset
          command: firewall-cmd --permanent --query-service=systemroletest
          register:  __result
          failed_when: __result is success
```

Reply:
I'll go and update this, thank you.